### ISSUE TITLE
destVI bugs

### ISSUE THREAD

**Issue by romain-lopez** [scverse/scvi-tools#1320]
We have identified several issues with the current version of the DestVI code:
1. (critical) the current code doesn't train at all on master. Can has flagged an issue with the nobs that needs to be resolved
2. (important, minor) there is an undesirable setup on the neural net architecture of the amortization network for the proportions (not for the gamma used throughout the paper)
3. (important, minor) there is some discrepancy in the treatment of the library size that must be figured out in between CondSCVI and DestVI

**Comment by cane11** [comment]
1 is fixed by branch fixed_elbo from Adam this morning
2 I will try with simulated data and open a branch afterwards
3 Currently running some tests and

**Comment by cane11** [comment]
Using the lymph node dataset from the preprint.

**Comment by giovp** [comment]
not sure if related but trying destVI on a new dataset following the tutorial, I get kernel error at this stage:
```
sc_model.train(
    max_epochs=250,
)
```
Other times instead, can't reproduce, I get nan in the encoder output (again trying to reproduce but can't rn). Few months ago it was working fine though. Will try if #1457 fix it.

**Comment by giovp** [comment]
ok, having the same issue with Stereoscope implementation, I'm afraid is independent. Am currently on a mac. Apologies for hijacking the issue.

**Comment by cane11** [comment]
The reason for the PR is not numerical instability but instability for estimated cell type proportions with two bug . If it is also an issue with Stereoscope, I would suggest to check the input data first (single dataset where this is happening) and you might want to reduce learning rate in CondSCVI

**Comment by adamgayoso** [comment]
This should have been closed by #1457
